sklearn/_loss/_loss.pyx.tp

```diff
@@ -1167,10 +1167,9 @@ cdef class CyHalfMultinomialLoss(CyLossFunction):
                     sum_exps = p[n_classes + 1]  # p[-1]
                     loss_out[i] = log(sum_exps) + max_value
 
-                    for k in range(n_classes):
-                        # label decode y_true
-                        if y_true[i] == k:
-                            loss_out[i] -= raw_prediction[i, k]
+                    # label encoded y_true
+                    k = int(y_true[i])
```

Comment:
**Not actionable**: I see how `y_true` ended up to be a float here, but I still find it strange.

Reply:
me too